WebDriverAgentLib/Utilities/FBImageIOScaler.m

```diff
@@ -122,18 +122,16 @@ - (nullable NSData *)scaledImageWithImage:(NSData *)image
                                     error:(NSError **)error
 {
   UIImage *uiImage = [UIImage imageWithData:image];
-  UIImageOrientation orientation = uiImage.imageOrientation;
-
   CGSize originalSize = uiImage.size;
-  Boolean shouldScale = fabs(1.0 - scalingFactor) > FLT_EPSILON;
-
+  BOOL shouldScale = fabs(1.0 - scalingFactor) > FLT_EPSILON;
   CGSize scaledSize;
   if (shouldScale) {
     scaledSize = CGSizeMake(originalSize.width * scalingFactor, originalSize.height * scalingFactor);
```

Comment:
This won't help us to save resources. I meant we should not call UIGraphicsBeginImageContext/drawInRect/UIGraphicsEndImageContext block if there is no need to scale

Reply:
I missed a comment. 
When I did like https://github.com/appium/WebDriverAgent/pull/488/commits/f7873ee00866d7f47b6487ff33dc47953a236105 (use `uiImage` without change), then landscape mode test failed.
Haven't seen the exact screenshot in landscape, so will check later.

Reply:
Hm... `UIImagePNGRepresentation(resultImage)` generates `(width = 828, height = 1792)` image even the given `resultImage` was `(width = 1792, height = 828)` in landscape without `UIGraphicsBeginImageContext`, `UIGraphicsGetImageFromCurrentImageContext` and `UIGraphicsEndImageContext`...
So, it seems we should keep them (can skip only `drawInRect`) to reflect orientation.